gsa/radix.go

```diff
@@ -1,20 +1,92 @@
 package gsa
 
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
 // Sort the string x using a count sort
 func CountSort(x string) string {
-	return ""
+	var sb strings.Builder
+	counts := make(map[byte]int)
+	keys := []int{}
+	//count occourences
+	for i := range x {
+		if _, ok := counts[x[i]]; !ok {
+			keys = append(keys, int(x[i]))
+		}
+		counts[x[i]]++
+	}
+
+	//IS THIS OKAY ??? k log k. Should still be constant or order 'len x' in normal cases?
+	sort.Ints(keys)
+
+	//create ordered string
+	for _, k := range keys {
+		v := counts[byte(k)]
+		for i := 0; i < v; i++ {
+			sb.WriteByte(byte(k))
+		}
+	}
+	return sb.String()
```

Comment:
You really don't need a builder here. You know the length of the result, so just allocate it

```go
  res := make([]byte, len(x))
```

Then run through the counts and fill `res` accordingly.